langlib/lang.regexp/src/main/java/org/ballerinalang/langlib/regexp/Find.java

```diff
@@ -109,9 +135,13 @@ private static void checkIndexWithinRange(BString str, long startIndex) {
         }
 
         int strLength = str.length();
-        if (strLength <= startIndex) {
+        if (strLength != 0 && strLength <= startIndex) {
             throw BLangExceptionHelper.getRuntimeException(BallerinaErrorReasons.INDEX_OUT_OF_RANGE_ERROR,
                     RuntimeErrors.INVALID_REGEXP_FIND_INDEX, startIndex, strLength);
         }
     }
+
+    private static boolean isEmptyRegexp(BRegexpValue regExp) {
```

Comment:
Move this to util class and reuse everywhere

Reply:
Updated in https://github.com/ballerina-platform/ballerina-lang/pull/39444/commits/f4b63bf37c6784813d85b782ccea40b35c7d319c.